Formula/git.rb

```diff
@@ -13,19 +13,25 @@ class Git < Formula
 
   option "with-blk-sha1", "Compile with the block-optimized SHA1 implementation"
   option "without-completions", "Disable bash/zsh completions from 'contrib' directory"
-  option "with-brewed-openssl", "Build with Homebrew OpenSSL instead of the system version"
-  option "with-brewed-curl", "Use Homebrew's version of cURL library"
-  option "with-brewed-svn", "Use Homebrew's version of SVN"
+  option "with-openssl", "Build with Homebrew's OpenSSL instead of using CommonCrypto"
+  option "with-curl", "Use Homebrew's version of cURL library"
+  option "with-subversion", "Use Homebrew's version of SVN"
+  option "with-perl", "Build against a custom Perl rather than system default"
   option "with-persistent-https", "Build git-remote-persistent-https from 'contrib' directory"
 
+  deprecated_option "with-brewed-openssl" => "with-openssl"
+  deprecated_option "with-brewed-curl" => "with-curl"
+  deprecated_option "with-brewed-svn" => "with-subversion"
+
   depends_on "pcre" => :optional
   depends_on "gettext" => :optional
-  depends_on "openssl" if build.with? "brewed-openssl"
-  depends_on "curl" if build.with? "brewed-curl"
+  depends_on "openssl" => :optional
+  depends_on "curl" => :optional
   depends_on "go" => :build if build.with? "persistent-https"
+  depends_on :perl => ["5.6", :optional]
   # Trigger an install of swig before subversion, as the "swig" doesn't get pulled in otherwise
   # See https://github.com/Homebrew/homebrew/issues/34554
-  if build.with? "brewed-svn"
+  if build.with? "subversion"
     depends_on "swig"
     depends_on "subversion" => "with-perl"
```

Comment:
Could maybe solve that like this:
```diff
diff --git a/Formula/git.rb b/Formula/git.rb
index b14e56f467..f23df082a2 100644
--- a/Formula/git.rb
+++ b/Formula/git.rb
@@ -28,8 +28,14 @@ class Git < Formula
   depends_on "openssl" => :optional
   depends_on "curl" => :optional
   depends_on "go" => :build if build.with? "persistent-https"
-  depends_on :perl => ["5.6", :optional]
-  depends_on "subversion" => :optional
+
+  if build.with? "subversion"
+    depends_on "subversion"
+    depends_on :perl => ["5.6", :recommended]
+  else
+    option "with-perl", "Build against a custom Perl rather than system default"
+    depends_on :perl => ["5.6", :optional]
+  end
 
   resource "html" do
     url "https://www.kernel.org/pub/software/scm/git/git-htmldocs-2.12.2.tar.xz"
```